### DISCUSSION TITLE
How to automatically fix font?

### DISCUSSION THREAD

**Discussion by hippylover** [fonttools/fonttools#3048] [Q&A]
I am using the Feorag https://www.feorag.com/freestuff/strangenewes.html font for a project im working on. When i use it in gimp and justify the text the bounding box cuts out part of some of the letters(for example "," "." and "f"). This might be a bug in gimp or in the font because it only happens with some fonts. However i want to fix this. I consequently opened the font in fontforge and tried to export it again and were barraged with bug messages. I click through the messages but there are so many and only some where fontforge can fix the problem automatically. Is there a way to fix the fonts automatically with fonttools(i already ran fonttools ttLib.removeOverlaps ../StrangeNewes.ttf StrangeNewes_removed_overlaps.ttf which reduced the file size and probably fixed a bunch of issues).

The problem is i want to fix the font but have a really hard time in investing the time in order to learn how to make a font first(probably invest a couple days to even know how to fix the font) so thats why i just want a bunch of commands to run. Non-integral coordinates. Self-intersecting, Missing points at extrema. are the errors im getting. I also tried to click through and click fix on all of them but some of them never disappear(the messages) even though i click "fix" a million times.

**Comment by jenskutilek** [comment]
I don't see anything wrong with the font. I think it's a GIMP bug.

Here's what it looks like in GIMP:

![Bildschirm­foto 2023-03-17 um 11 36 46](https://user-images.githubusercontent.com/3185950/225882010-dc464489-6634-4ebc-99d2-be587460d5b6.png)

The comma at least is fully inside the glyph's sidebearings:

![Bildschirm­foto 2023-03-17 um 11 38 35](https://user-images.githubusercontent.com/3185950/225882089-7dd92c01-d3b8-4493-91d2-41c631139ccf.png)

**Comment by hippylover** [comment]
Yeah i saw that too. but there are a bunch of errors so perhaps gimp and the font "collaborates" into showing as a bug. i ended up using another program for the project as i see this has been a bug in gimp for several years already.

**Reply by ftCLI** [reply]
This is a screenshot from FontValidator check.

![image](https://user-images.githubusercontent.com/83063506/225970750-2088071d-c421-431e-9a21-5caa6231a2f6.png)

Find attached a file without [LTSH]([StrangeNewes.zip](https://github.com/fonttools/fonttools/files/11003942/StrangeNewes.zip)) table to test; [StrangeNewes.zip](https://github.com/fonttools/fonttools/files/11003942/StrangeNewes.zip)

P.S.: RemoveOverlaps reduced the file size because it dropped hinting.
